### PR TITLE
Experimental pipelines

### PR DESCRIPTION
**Pull request by mostafa**
The details are in the README as part of updates.

@thomaspatzke @kelnage I'd be happy to have your review.

Related to:
- #110

### CODE REVIEW

File: sigma/plugins.py [hunk 1]
```diff
@@ -90,60 +150,74 @@ def _discover_module_directories(cls, module, directory_name : str, include : bo
         return result
 
     @classmethod
-    def autodiscover(cls, include_backends : bool = True, include_pipelines : bool = True, include_validators : bool = True):
```
Comment (mostafa) [inline sigma/plugins.py:93]: I ran `black` on this file only, which resulted in the changes from here on.

Comment (thomaspatzke) [reply sigma/plugins.py:93]: The resulting code is much more readable, like it. But lets keep it back a while for the rest of the code. I'm currently implementing a new feature with some files touched and fear the merge conflicts 😉

**Comment by thomaspatzke** [comment]
I ♥️ it! Solves one of the things I really disliked in an elegant way. Thanks for the contribution!

File: sigma/plugins.py (hunk 1 above)
Comment (mostafa) [reply sigma/plugins.py:93]: No rush!

**Comment by mostafa** [comment]
@thomaspatzke Glad you liked it.

**Pull request merged**